src/extent2.c

```diff
@@ -194,6 +197,35 @@ extents_alloc(tsdn_t *tsdn, arena_t *arena, ehooks_t *ehooks, ecache_t *ecache,
 	return edata;
 }
 
+edata_t *
+extents_alloc_grow(tsdn_t *tsdn, arena_t *arena, ehooks_t *ehooks,
+    ecache_t *ecache, void *new_addr, size_t size, size_t pad, size_t alignment,
```

Comment:
ecache isn't being used. Is that intentional?

Reply:
No; follow-on diffs were going to try to break the arena dependency here, but I thought this stack of commits was already getting a little bit big. I'll remove it.

Reply:
Ah I see. Feel free to keep it then.